.github/workflows/qa.yaml

```diff
@@ -90,17 +90,33 @@ jobs:
         run: |
           set -eu
           cargo install grcov
-      - name: Run tests
+      - name: Run tests (with coverage collection)
         run: |
           set -eu
 
-          ./gotestcov ./...
+          # The coverage is not written if the output directory does not exist, so we need to create it.
+          raw_cov_dir="/tmp/raw_files"
+          rm -fr "${raw_cov_dir}"
+          mkdir -p "${raw_cov_dir}"
+
+          # Overriding the default coverage directory is not an exported flag of go test (yet), so
+          # we need to override it using the test.gocoverdir flag instead.
+          #TODO: Update when https://go-review.googlesource.com/c/go/+/456595 is merged.
+          go test -cover -covermode=set ./... -shuffle=on -args -test.gocoverdir="${raw_cov_dir}"
+
+          # Convert the raw coverage data into textfmt so we can merge the Rust one into it
+          go tool covdata textfmt -i="${raw_cov_dir}" -o="/tmp/coverage.out"
+
+          # Append the Rust coverage data to the Go one
+          cat "${raw_cov_dir}/rust-cov/rust2go_coverage" >>"/tmp/coverage.out"
+
+          # Filter out the testutils package and the pb.go file
+          grep -v -e "testutils" -e "pb.go" "/tmp/coverage.out" >"/tmp/coverage.out.filtered"
+
       - name: Run tests (with race detector)
         run: |
           go test -race ./...
       - name: Upload coverage to Codecov
         uses: codecov/codecov-action@v3
         with:
-          file: coverage/coverage.out.filtered
-
-  # TODO: rust-tests:
```

Comment:
Looks like a sneaky change, but as we are touching the action already and this is definitely related to the tests, I think it makes sense... I don't think we will be writing unit tests for the Rust side, as it is simple and (very) reliant on the gRPC server to be running... (I can revert this if I'm wrong, though)